awx/api/serializers.py

```diff
@@ -4767,6 +4768,28 @@ def validate(self, attrs):
         return super(ScheduleSerializer, self).validate(attrs)
 
 
+class InstanceLinkSerializer(BaseSerializer):
+    class Meta:
+        model = InstanceLink
+        fields = ('source', 'target')
+
+    source = serializers.SlugRelatedField(slug_field="hostname", read_only=True)
+    target = serializers.SlugRelatedField(slug_field="hostname", read_only=True)
+
+
+class InstanceNodeSerializer(BaseSerializer):
+    class Meta:
+        model = Instance
+        fields = ('id', 'hostname', 'node_type', 'node_state')
+
+    node_state = serializers.SerializerMethodField()
+
+    def get_node_state(self, obj):
+        if not obj.enabled:
+            return "disabled"
+        return "unhealthy" if obj.errors else "healthy"
```

Comment:
We don't check for the health of hop nodes. Seeing this, I realize that we should.

To outline that in better clarity, if the `receptorctl` data shows that a hop node has failed to check in within the configured receptor broadcast frequency (practically, 60 seconds), then it should have some errors filled in. Likewise, the errors should be cleared if it appears on the mesh afterwards.

Could you file an issue for that and add to the project so that we make sure to cover it? The purpose of doing this would be strictly to make the mesh graph data accurate. Thanks.